### DISCUSSION TITLE
Help needed : error message when using MultilevelIV

### DISCUSSION THREAD

**Discussion by pinson06** [mmeierer/REndo#62] [Q&A] [answered]
Hi everyone,

I'm trying to use this very nice package but I obtain a error message for which I have no clue how to solve. Here's what i obtain in R when using the MultiLevelIV with my dataset and this code : 
formula <- y ~ e + a + f + w + l + w2 + w_l + f_l + e_l + e_w + a_w + a_l + a_e + e_w_l +   f_w_l + f_a_l + f_e_l + a_w_l + f_a_e_l + v + pr + t + (1 | album_id) | endo(e,a,f)
multilevel <- multilevelIV(formula = formula, data = data, verbose=TRUE)

Message obtained via R's console:
"Detected multilevel model with 2 levels.
For album_id (Level 2), 181 groups were found.
Error: Matrices must have same dimensions in diag2Tsmart(e1, e2, "d") - e2"

Any ideas?

**Comment by pschil** [comment]
Hi,

thanks for using the package and for reporting! 
Its a little difficult to troubleshoot this from far without any data. We do not use the function `diag2Tsmart` anywhere directly so this seems to crash as part of an other operation. If matrices do not have the same dimension, this is often because some cols/rows were lost along the way because of data issues.

Can you maybe check whether: 
- there are any NAs in the data
- there are any special characters (such as ñ, č,~, <, 'space', etc) in the data, especially in the `album_id` column
- you can fit `lmer()` of the `lme4` package just on the first part of the formula excluding the `endo()` special and see whether any observations were lost (ie fit lmer with only `y ~ e + a + f + w + l + w2 + w_l + f_l + e_l + e_w + a_w + a_l + a_e + e_w_l + f_w_l + f_a_l + f_e_l + a_w_l + f_a_e_l + v + pr + t + (1 | album_id)`)

**Comment by pinson06** [comment]
Hi Patrick,

Thanks for your help! 
Unfortunately, I checked the different elements that you mentioned: there is no NA, nor special characters in my data.frame. Moreover, no observation is lost when using lmer().

Here's a sample of the dataset, maybe it can help:

[data sample rendo apb.xlsx](https://github.com/mmeierer/REndo/files/7014726/data.sample.rendo.apb.xlsx)

**Comment by pschil** [comment]
Thanks again for reporting and providing example data. 

I first thought this might be because the data is rank-deficient but there is actually bug in the code which was revealed by having only two same-sized groups. Let me turn this into an issue...

**Comment by pschil** [comment, marked as answer]
I have fixed the issue but it currently cannot be released to CRAN because the submission team is on vacation. 
You can however install the latest source package from github using:

```
remotes::install_github(repo="mmeierer/REndo", ref="release-2.4.3")
```

**Reply by pinson06** [reply]
Wow, thanks a lot, it now works ok :)

**Reply by mmeierer** [reply]
Thanks @pschil for addressing this issue in such a speedy way!
